src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,290 @@
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import inquirer from 'inquirer';
+import fetch from 'node-fetch';
+import { Dictionary } from '../../dependency-manager/utils/dictionary';
+import { ComponentSpec } from '../../';
+import { ServiceSpec } from '../../dependency-manager/spec/service-spec';
+
+const download = require('download-git-repo');
+
+interface GitHubRepo {
+  name: string,
+  full_name: string,
+}
+
+interface Selection {
+  name: string,
+  architect_yml: Dictionary<any>,
+  depends_on: string,
+}
+
+const GITHUB_BRANCH = 'main';
+
+class Queue<T> {
+  items: any[];
+
+  constructor(...params: any[]) {
+    this.items = [...params];
+  }
+
+  enqueue(item: T) {
+    this.items.push(item);
+  }
+
+  dequeue() {
+    return this.items.shift();
+  }
+
+  getItems() {
+    return this.items;
+  }
+
+  hasItem(key: string) {
+    return this.items.some(item => item.hasOwnProperty(key));
+  }
+
+  size() {
+    return this.items.length;
+  }
+}
+
+export default class ProjectUtils {
+  static queue = new Queue();
```

Comment:
Static like this should be avoided. The intent is good, but the issue is that this Util class now has state that must be managed. If we are going to add state we should generally do 1 of 2 things.
1. Create a new class that stores this as instance variables and hide most of the logic away as private functions.
2. Create a data class that stores the intermediary state necessary for the operation to work and pass that through each function.

In this instance, I would opt for number 1 and create a new src folder under `src/common` called project-generation.